### PR TITLE
Update DG and UG

### LINKED ISSUE #214: [PE-D][Tester E] Adding a person with the "s/o" or "d/o" title in their name is not supported.
Entered command: add n/EG NAME AS LONG AS I WANT s/o NAME p/92365612 e/kdsfsdsdf@smth.com a/fdsfsdfsdfsdfsd t/HAHA

Expected: Users with entering people with "s/o" or "d/o" in their names would expect for this to be a valid add entry.

Actual: The error message that pops up is: "Names should only contain alphanumeric characters and spaces, and it should not be blank."

Thus, adding a person with the "s/o" or "d/o" title in their name is not supported, which indicates the add command support is incomplete.

-------------
Labels: `type.FeatureFlaw` `severity.Medium`
original: youngseopark05/ped#1

### LINKED ISSUE #231: [PE-D][Tester D] No warpping for very long tags
-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: slidings/ped#8

### LINKED ISSUE #277: [PE-D][Tester C] The app accepts email with invalid domain name
Describe the bug:
The app is not stopping users from adding a client when the email is inputted with an invalid domain , i.e. `test@123`

Steps to reproduce:
1. Type `add n/Jolyn p/12345678 e/test@123 a/Jolyn Home` and hit enter

Expected behaviour:
Appropriate error message displayed once user hits enter

Actual behaviour:
The client is added successfully

Screenshot:

Potential problematic behaviour:
This might be caused by mistyping by a user, and if the user is not aware of this, when the user trying to get the email of the client to contact them, there will be a problem 

-------------
Labels: `type.FeatureFlaw` `severity.Medium`
original: ruiming97/ped#10

### PR DESCRIPTION
**Pull request by RezwanAhmed123**
Let's update the UG and DG to ensure consistent documentation and user guide for users.

Let's also include more planned enhancements such that it helps give some immunity during the PE.

Fix #277 ,
Fix #231 ,
Fix #214 .

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -169,7 +170,21 @@ the same full name with different contact details but it will be unreasonably ra
...
+Command keywords are currently case-sensitive i.e. `add` is a valid command but `Add` is not. For user convenience, we
+will make command keywords case-insensitive in a future update by parsing the prefixes differently.
+
+### 5. No Wrapping for very long ui details
+Currently, if the name, tags or other details are very long, they are cut off in the UI. This will be fixed in a future
+version by enabling wrapping for texts in the UI such that they are more easily viewable.
+
+### 6. Some common names are not allowed to be entered
+Currently, our app does not allow some common Singaporean names with some special characters to be entered. An example
+is "s/o" or "d/o". This will be fixed in a future version because we will change the way we parse the add and edit
```
Comment (matthewyip1511) [inline docs/DeveloperGuide.md:182]: Maybe we can add that:

In the future, we will add support for more name formats which include, but are not limited to:

1. Names with a `/` eg. `s/o` `d/o`
2. Names with an apostrophe eg. `D'Silva`
3. Names with a hyphen eg. `John Lim-Tan`
4. Names with an @ eg. `Tan Xiao Ming @Chen Xiaoming`

To increase the robustness of this planned enhancement 🙂

**Review by matthewyip1511 — CHANGES_REQUESTED** [review]
LGTM other than for the change mentioned 👍

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (RezwanAhmed123) [reply docs/DeveloperGuide.md:182]: if it becomes this specific, it can be argued that it is indeed 4 separate enhancements

**Review by matthewyip1511 — APPROVED** [review]
LGTM! Merging

**Pull request merged**